### PR TITLE
Add a CI job to build a special version for translation reviewers 

### PR DESCRIPTION
**Pull request by loremattei**
This PR adds a CI job triggered by an API call that builds a custom version of the app for translations reviewers. 
The build process downloads all the pending translations, merges them into the current ones, builds the app and uploads it to Appetize.

To test:
1. Get a CircleCI personal token. 
2. From the terminal, run the following curl command to start the build:
```
curl -u <token>: -X POST --header "Content-Type: application/json" -d '{
 "branch": "translation-reviewers-build",
 "parameters": {
   "translation_review_build": true
 }
}' https://circleci.com/api/v2/project/github/wordpress-mobile/WordPress-Android/pipeline
```
3. Verify that the build finishes without errors.
4. Go to Appetize.io and verify that the app runs as expected.

**Note**: Before merging this, a new version of the `release-toolkit` has to be released and the reference in this PR updated. Related PR: https://github.com/wordpress-mobile/release-toolkit/pull/133. 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by oguzkocer** [comment]
Looks like @jkmassel is on this PR, so I am taking myself off. Please feel free to add me again if you want me to review it as well.

**Comment by loremattei** [comment]
Ah! Good catch @jkmassel! It looks like I actually merged develop to fix a merge conflict, but I missed a change that was made on the Gutenberg build step on CircleCI and that I had to add to my new piece as well.

File: Gemfile.lock [hunk 1]
```diff
@@ -1,9 +1,9 @@
 GIT
   remote: https://github.com/wordpress-mobile/release-toolkit
-  revision: 6c1fa45f3beb216f4f5d03346f2981f1024b799e
-  tag: 0.9.6
+  revision: 4c2b3b0403979b4d6c8485bbd54f27031e8f795e
+  branch: traslator-review-build
```
Comment (mokagio) [inline Gemfile.lock:4]: Leaving this comment here just so we remember to point back to a released version before merging the PR 😉

File: fastlane/Fastfile [hunk 2]
```diff
@@ -453,6 +453,30 @@ ENV["HAS_ALPHA_VERSION"]="true"
...
+  # -----------------------------------------------------------------------------------
+  # This lane builds an app bundle with pending translations
+  # -----------------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane build_for_translation_review [custom_version:<custom_version>]
+  #####################################################################################
+  desc "Builds an app bundle"
+  lane :build_for_translation_review do | options |
+    cmd_params = ""
+    if (!options[:custom_version].nil?)
```
Comment (mokagio) [inline fastlane/Fastfile:467]: Nitpick

```suggestion
    unless options[:custom_version].nil?
```

File: fastlane/Fastfile [hunk 3]
```diff
@@ -453,6 +453,30 @@ ENV["HAS_ALPHA_VERSION"]="true"
...
+      UI.message("Building custom version: #{options[:custom_version]}") 
+      cmd_params=" -PversionName=\"#{options[:custom_version]}\""
+    end
+
+    sh("../tools/update-translations.sh")
+    sh("../tools/update-translations.sh waiting")
+    sh("../tools/update-translations.sh fuzzy")
+
+    android_merge_translators_strings(strings_folder:"./WordPress/src/main/res/")
+    sh("export SUPPRESS_GUTENBERG_MOBILE_JS_BUNDLE_BUILD=1 && cd .. && ./gradlew --stacktrace assembleJalapenoDebug #{cmd_params}")
```
Comment (mokagio) [inline fastlane/Fastfile:477]: Question out of curiosity, would this work with this syntax?
```suggestion
    sh("cd .. && SUPPRESS_GUTENBERG_MOBILE_JS_BUNDLE_BUILD=1 ./gradlew --stacktrace assembleJalapenoDebug #{cmd_params}")
```

**Pull request merged**